packages/realm/src/schema/normalize.ts

```diff
@@ -116,7 +116,7 @@ export function normalizeObjectSchema(arg: RealmObjectConstructor | ObjectSchema
 
   const { name, primaryKey, asymmetric, embedded, properties } = arg;
   assert(name.length > 0, "Invalid schema for unnamed object: 'name' must be specified.");
-  const primaryKeyFieldIsMissing = primaryKey && !Object.hasOwn(properties, primaryKey);
```

Comment:
Any particular reason for this change? Do you fear it won't be available on the older versions of JSC we support? If so, we should consider downgrading the `target` & `libs` of the ts-config.json.

Reply:
Not just _fear_ that it won't be available. It _isn't_ available and tests were failing without this change.

I'm not sure what should be responsible for doing our polyfilling. It feels a bit like something that should be handled by metro bundler of the whole application when it knows the precise deployment target rather than just the library. In some cases we _do_ want to use newer features than are available everywhere as long as they are available wherever they are used.

Reply:
Also, it seems like JSC is a low-priority target. So we may be OK with saying that some new features won't work there, although "core functionality" like this still must. I don't know if that is possible to achieve with the ts config, but I think as long as we get our CI to include the JSC versions we care about, I think we will be fine in practice either way.

Reply:
It would probably make sense to spend some more time on this, after this PR is merge.
I agree its not a high priority if we're actually testing on the lowest supported JSC version, but again - I would also like to fail faster (ideally in the editor).

If we truly want to support older runtimes we should either
- transform our code (using babel?) and potentially produce a different bundle for old JSC targets (if those is practical to distinguish from newer ones) or
- make sure we're not using the newest and greatest features of the language (which would be a bummer and sort of the direction this change is taking).

Reply:
> transform our code (using babel?) and potentially produce a different bundle for old JSC targets (if those is practical to distinguish from newer ones) 

Doesn't the metro bundler already do that? I think it knows if it knows if it is targeting hermes or JSC. Maybe we just need to publish a recommended config for users on JSC?

> make sure we're not using the newest and greatest features of the language (which would be a bummer and sort of the direction this change is taking).

I agree that this probably isn't the direction we want to take. But in this case the new feature is only slightly nicer than the old one, so it wasn't worth figuring out a general solution for. I'm not trying to set a precedent here, just trying to get us working on JSC. 😄